modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryManager.java

```diff
@@ -325,6 +326,15 @@ public long memoryLimit() {
         return globalQuota;
     }
 
+    /**
+     *
+     * @return memory in bytes currently left available for the queries on this node.
+     */
+    public long getFreeMem() {
+        return ((LongMetric)ctx.metric().registry(SqlMemoryStatisticsHolder.SQL_QUOTAS_REG_NAME).findMetric("freeMem")).value();
+    }
+
```

Comment:
Why SqlMemoryManager refers to MetricRegistry for a metric?
SqlMemoryManager is a metric supplier, and JMX bean should refers to a registry, then the registry should ask memory manager for a calculated "freemem".